topics/icrc_X_multiple_canister_call.md

```diff
@@ -0,0 +1,237 @@
+
+5. The relying partner, if any of the transactions failed, implements error handling. The response of the individual transactions will be aggregated into the response of the batch call. If there are any errors in the response, it is up to the relying party to decide how to handle the error. 
+
+**Flow**
+
+```mermaid
+sequenceDiagram
+    participant RP as Relying Party
+    participant S as Signer
+    participant U as User
+    participant C as Target Canister
+
+    RP ->> S: Request batch canisters call
+    alt Relying party has not been granted the `icrcX_call_canister` permission scope<br>or the request does not comply with scope restrictions
+        S ->> RP: Error response: Permission not granted (3000)
+    else
+
+        alt If signer want to use consent message from canisters
+            loop for each call of calls in the `icrcX_call_canister`
+                S ->> C: Get the consent message
+                C ->> S: Return consent message
+            end
+        end
+
+        S ->> U: Showing signing message from consent message or blind message
+        
+
+        alt Approved
+            U ->> S: Approve request
+            Note over S,C: Call the request for each canister in one of following two ways: <br/> - `parallel` <br /> - `sequence` abort if result is not success
+                %% this is a comment 
+                
+                S ->> C: Submit canister call
+                S ->> S: Wait for the canister call result
+                S ->> S: Add call result to batch call response
+            Note over S,C: End of the execution
+        
+        S ->> U: Display success/failed message
+        S ->> RP: Return the batch call response
+            
+        else Rejected
+            U ->> S: Reject request
+            S ->> U: Display reject message
+            S ->> RP: Error response: Action aborted (3001)
+        end
+    end
+```
+
+### Examples
+
+#### Successful Batch Call
+
+Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrcX_batch_call_canisters",
+  "params": {
+    "mode": "sequence",
+    "sender": "b7gqo-ulk5n-2kpo7-oalt7-p2kyl-o4j5l-kiuwo-eeybr-dab4l-ur6up-pqe",
+    "requests": [
+      {
+        "id": 1,
+        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+        "method": "transfer",
+        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+      },
+      {
+        "id": 2,
+        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+        "method": "transfer",
+        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+      }
+    ]
+  }
+}
+```
+
+Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "responses": [
+      {
+        "id": 1,
+        "result": {
+          "Ok": "16509194"
```

Comment:
```suggestion
          "contentMap": "2dn3p2NhcmdYTkRJREwEbXtuAGwCs7DawwNorYbKgwUBbAP7ygECot6U6wYB2KOMqA19AQMBHVfs7SoKuxOJdX37k7sBJsPALWEo5ayflizuaH0CAADIAWtjYW5pc3Rlcl9pZEoAAAAAAcDR1wEBbmluZ3Jlc3NfZXhwaXJ5GxeNX/65y4YAa21ldGhvZF9uYW1laHRyYW5zZmVyZW5vbmNlUFF4+hAimFhoqkdUcIchz0xscmVxdWVzdF90eXBlZGNhbGxmc2VuZGVyWB1q63Snu+4C5/fpWFu4nq1IpZxCYDEYA8XSPqPfAg==",
          "certificate": "2dn3omR0cmVlgwGDAYIEWCAPzKZJY/emKhi2GGtBrnHh4cdttATd4+9GtJrNCBepb4MBgwJOcmVxdWVzdF9zdGF0dXODAYIEWCCCgynUaonrKCCywghWCSk9BeDqMoI4yf15nxyU/5JZv4MBggRYIDG7WdzQ9sGWI1MpxizUzxubsEBuNkTT94UOZ9USbzNvgwGCBFggawwbTHxnPUzBAUhWBRjk0nzPs2fPpJlaIYtj5AvcX+ODAYIEWCDiFLyaWuMWjtVurCQcSgny/cqfM8S6qrdihVq7nPz1FoMCWCD/8jdeccvqHVYf06Hw7qPXIDNimC1Uyf47VsvgqKpPiIMBgwJFcmVwbHmCA1RESURMAWsCvIoBfcX+0gFxAQAABIMCRnN0YXR1c4IDR3JlcGxpZWSCBFgg7qZngcNt2+B/RuF44W3LRsKWXG6QQg2L6GdZgJ6Nb3+DAYIEWCAx3tU/mhHfX+wDzF003eSJYN8Nebou8rTeGyxr/rUa1YMCRHRpbWWCA0nw9+r88fjXxhdpc2lnbmF0dXJlWDCXNshvwWG1jGViP7ELePGHCThBw9mts45FxIy4gZATkUEsPeJ6y+cjbn2REmB0Soo="
```

As shown [here](https://github.com/dfinity/wg-identity-authentication/blob/main/topics/icrc_49_call_canister.md#example-rpc-response) in ICRC-49, the signer can't and shouldn't decode the response here since it doesn't know the response encoding and the relying party needs the original certificate and contentMap to verify the call has actually been made as requested.